internal/controllers/topology/cluster/structuredmerge/options.go

```diff
@@ -18,6 +18,44 @@ package structuredmerge
 
 import (
 	"sigs.k8s.io/cluster-api/internal/contract"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+var (
+	// defaultAllowedPaths are the allowed paths for all objects except Clusters.
+	defaultAllowedPaths = []contract.Path{
+		// apiVersion, kind, name and namespace are required field for a server side apply intent.
+		{"apiVersion"},
+		{"kind"},
+		{"metadata", "name"},
+		{"metadata", "namespace"},
+		// the topology controller controls/has an opinion for labels, annotation, ownerReferences and spec only.
+		{"metadata", "labels"},
+		{"metadata", "annotations"},
+		{"metadata", "ownerReferences"},
+		{"spec"},
+	}
+
+	// defaultAllowedPathsCluster are the allowed specific for Clusters.
```

Comment:
```suggestion
	// allowedPathsCluster are the allowed paths specific for Clusters.
```
I would drop the default as I would only consider the ones above as default now given that a consumer of the helper can't overwrite it anymore